python/dolfin/fem/assembling.py

```diff
@@ -114,7 +114,7 @@ def assemble_system(A_form,
 
     fem.assemble(b_tensor, b_form)
     fem.apply_lifting(b_tensor, [A_form], [bcs])
-    b_tensor.ghostUpdate(addv=PETSc.InsertMode.ADD, mode=PETSc.ScatterMode.REVERSE)
+    b_tensor.ghostUpdate(addv=PETSc.InsertMode.INSERT, mode=PETSc.ScatterMode.FORWARD)
```

Comment:
I think this is wrong. We want to accumulate ghost contributions onto the owning processes.

Reply:
Hmm, if no BCS are specified (as is usually the case for projection) then ghost update with `ADD` will change the `b_tensor`. It will gather and accumulate values for ghost positions. This test will fail for ADD.

But you're right `INSERT` is also wrong.

Reply:
Changing the `b_tensor` is required also for no bcs, and it should be `ADD`. 

What's wrong here is `fem.assemble(b_tensor, b_form)`. It should be `assemble_vector`.

Reply:
Thanks, fixed now.